goalie/function_data.py

```diff
@@ -280,6 +280,53 @@ def _export_h5(self, output_fpath, export_field_types, initial_condition=None):
                             f = self._data[field][field_type][i][j]
                             outfile.save_function(f, name=name, idx=j)
 
+    def transfer(self, target, method="interpolate"):
+        """
+        Interpolate or project all functions from this :class:`~FunctionData` object to
+        the target :class:`~FunctionData` object.
+
+        :arg target: the target :class:`~FunctionData` object to which to transfer the
+            data
+        :type target: :class:`.FunctionData`
+        :arg method: the transfer method to use, either 'interpolate' or 'project'
+        :type method: :class:`str`
+        """
+        stp = self.time_partition
+        ttp = target.time_partition
+
+        if method not in ["interpolate", "project"]:
+            raise ValueError(
+                f"Transfer method '{method}' not supported."
+                " Supported methods are 'interpolate' or 'project'."
+            )
```

Comment:
I wonder if it'd make sense to support `prolong`? That way we could use this method to transfer fields to the enriched space?

Reply:
I added this in https://github.com/mesh-adaptation/goalie/pull/243/commits/3952b60e2bd32789fb99d49a82f694a4aaadbc0f but I don't think we can use it in `indicate_errors` since we only define one `Function` and we just overwrite its data